include/aikido/rviz/TrajectoryMarker.hpp

```diff
@@ -0,0 +1,137 @@
+  TrajectoryMarker(
+      interactive_markers::InteractiveMarkerServer* markerServer,
+      const std::string& frameId,
+      trajectory::ConstTrajectoryPtr trajectory,
+      const dart::dynamics::MetaSkeleton& skeleton,
+      const dart::dynamics::Frame& frame,
+      const Eigen::Vector4d& rgba = Eigen::Vector4d::Constant(0.75),
+      double thickness = 0.01,
+      std::size_t numLineSegments = 16u);
+
+  /// Destructor
+  ~TrajectoryMarker();
+
+  /// Sets or updates trajectory to visualize
+  ///
+  /// \param[in] trajectory C-sapce (or joint-space) trajectory. The statespace
+  /// of the trajectory should be MetaSkeletonStateSpace. Otherwise, throw
+  /// invalid_argument exception.
+  void setTrajectory(trajectory::ConstTrajectoryPtr trajectory);
+
+  /// Returns trajectory associated with this marker
+  trajectory::ConstTrajectoryPtr getTrajectory() const;
+
+  /// Sets or updates color (RGB) of visualizing trajectory.
+  void setColor(const Eigen::Vector3d& rgb);
+
+  /// Returns color (RGB) of visualizing trajectory.
+  Eigen::Vector3d getColor() const;
+
+  /// Sets or updates alpha of visualizing trajectory.
+  void setAlpha(double alpha);
+
+  /// Returns alpha of visualizing trajectory.
+  double getAlpha() const;
+
+  /// Sets or updates RGBA of visualizing trajectory.
+  void setRGBA(const Eigen::Vector4d& rgb);
+
+  /// Returns RGBA of visualizing trajectory.
+  Eigen::Vector4d getRBGA() const;
+
+  /// Sets or updates thickness of visualizing trajectory.
+  void setThickness(double thickness);
+
+  /// Returns thickness of visualizing trajectory.
+  double getThickness() const;
+
+  /// Sets or updates number of line segments of the visualizing trajectory.
+  void setNumLineSegments(std::size_t numLineSegments);
+
+  /// Returns number of line segments of the visualizing trajectory.
+  std::size_t getNumLineSegments() const;
+
+  /// Updates this marker.
+  ///
+  /// This function should be called after the properties are changed (e.g.,
+  /// trajectory, color, thickness, number of line-segments) so that RViz
+  /// reflects the changes accordingly.
+  void update();
+
+private:
+  /// Updates trajectory points based on trajectory and number of line-segments.
+  ///
+  /// This function is called by update().
+  void updatePoints();
+
+  /// Returns marker.
+  visualization_msgs::Marker& getMarker();
+
+  /// Returns const marker.
+  const visualization_msgs::Marker& getMarker() const;
+
+  /// RViz marker server.
+  interactive_markers::InteractiveMarkerServer* mMarkerServer;
+
+  /// RViz interactive marker.
+  visualization_msgs::InteractiveMarker mInteractiveMarker;
+
+  ///
+  std::string mFrameId;
+
+  /// C-sapce (or joint-space) trajectory.
+  trajectory::ConstTrajectoryPtr mTrajectory;
+
+  /// Number of line segments of the discretized task-space trajectory.
+  std::size_t mNumLineSegments;
+  // TODO(JS): Consider switching to resolution once arc-length calculation is
```

Comment:
What if we use a time resolution (which exists) rather than an arc-length resolution (which I guess doesn't yet exist)? I think the number of line segments is unintuitive.

Reply:
I would prefer discretizing a smooth curve based on its length so that we don't see giant straight lines for a long curve (The trajectory length computation will be introduced in implementing TrajOpt).

Using line segments might not be the best alternative for now but I believe this trick is widely used in practice. For example, GLU functions use similar parameters to draw a sphere, a cylinder, etc.

I actually would like to represent time resolution in a different visualization way (e.g., using different colors or alpha per seconds) as a future improvement.